src/internal/connector/exchange/event_calendar_cache.go

```diff
@@ -38,68 +37,54 @@ func (ecc *eventCalendarCache) Populate(
 	}
 
 	var (
-		asyncError  error
-		directories = make(map[string]graph.Container)
-		errUpdater  = func(s string, e error) {
-			asyncError = support.WrapAndAppend(s, e, err)
-		}
+		errs        error
+		directories = make([]graph.Container, 0)
 	)
 
-	query, err := ecc.gs.Client().UsersById(ecc.userID).Calendars().Get(ctx, options)
-	if err != nil {
-		return errors.Wrap(err, support.ConnectorStackErrorTrace(err))
-	}
+	builder := ecc.gs.Client().UsersById(ecc.userID).Calendars()
 
-	iter, err := msgraphgocore.NewPageIterator(
-		query,
-		ecc.gs.Adapter(),
-		models.CreateCalendarCollectionResponseFromDiscriminatorValue,
-	)
-	if err != nil {
-		return err
-	}
-
-	cb := IterativeCollectCalendarContainers(
-		directories,
-		"",
-		errUpdater,
-	)
+	for {
+		resp, err := builder.Get(ctx, options)
+		if err != nil {
+			return errors.Wrap(err, support.ConnectorStackErrorTrace(err))
```

Comment:
this may be something we need to revisit later but I'm not completely sure. The code overall looks fine, but I'm wondering how this will function if we get some transient error in the request. In that case, we may want to take some special action instead of returning an error immediately. However, that is something that can be handled in a different PR (and after we have more insight into how errors from calls like this appear if we don't already know since we've been using msgraph iterators thus far)

Reply:
@ashmrtn updates have been made to the description. Your approach to the delta operation iteration inspired this approach. It was discussed that the iterator framework carries a significant overhead to lines of code. Thus, prior to moving into the eventID section described as `Stage 2` needed to ensure that the approach to iteration aligns with your thoughts on what the overall framework will look like.

As for abstraction, the `Get()` functions called by the builders and the builders themselves make it difficult to abstract it to an `exchange.Iterator`. However, doing this inline for the resolvers is not a lot of overhead.